source/extensions/filters/http/grpc_json_transcoder/json_transcoder_filter.cc

```diff
@@ -897,10 +899,12 @@ bool JsonTranscoderFilter::maybeConvertGrpcStatus(Grpc::Status::GrpcStatus grpc_
   }
 
   std::string json_status;
-  auto translate_status =
-      config_.grpc_status_json_marshaller()
-          ? config_.grpc_status_json_marshaller()->marshal(*status_details, json_status)
-          : per_route_config_->translateProtoMessageToJson(*status_details, &json_status);
+  std::unique_ptr<Protobuf::Message> transformed_grpc_status =
+      config_.grpc_status_proto_transformer() && status_details
+          ? config_.grpc_status_proto_transformer()->transform(*status_details)
+          : nullptr;
+  auto translate_status = per_route_config_->translateProtoMessageToJson(
```

Comment:
How can we make sure translateProtoMessageToJson() is able to translate the proto message into JSON?  Does it have the proto-descriptor to do that translation?

Reply:
Update transcoder API to emphasize that descriptor should include the custom error proto type just like the error  details.